### ISSUE TITLE
Plans for edifact-lib project

### ISSUE THREAD

**Issue by hepabolu** [jono-tt/edifact-lib#1]
Hi, I found this project this weekend and found it useful for a project I'm working on. I noticed there is not much activity lately. 
Can you indicate what your plans are for this project? Are you going to keep it in active development or was it merely an exercise of some sort?

**Comment by jono-tt** [comment]
Hi,

We currently use it in production and have recently been asked to extend it support repetition characters. We would be really happy if you’d like to participate in its future development as we have a lot of EDI based communications coming up and so having other involved would be fantastic.

Please let me know.

Thanks,
Jono


On 23 Jan 2017, at 12:14, hepabolu <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi, I found this project this weekend and found it useful for a project I'm working on. I noticed there is not much activity lately.
Can you indicate what your plans are for this project? Are you going to keep it in active development or was it merely an exercise of some sort?

**Comment by hepabolu** [comment]
I can surely try to help. I'm not very 'proficient' in EDIFACT, and I need it for a very specific set of codes, so I'm not sure how useful I can be for your project.

I have made a small code change to allow for recursive definitions. If you're interested I can create a pull request.

**Comment by jono-tt** [comment]
Yes please do send a pr. the main reason for the library is exactly the same reason, EDI is terrible to work with so having a "standard" way of read, writing and validating it is very useful.

If you could also add your new schemes that would be great.


On 23 Jan 2017, at 17:20, hepabolu <notifications@github.com<mailto:notifications@github.com>> wrote:


I can surely try to help. I'm not very 'proficient' in EDIFACT, and I need it for a very specific set of codes, so I'm not sure how useful I can be for your project.

I have made a small code change to allow for recursive definitions. If you're interested I can create a pull request.

-

**Comment by hepabolu** [comment]
I've already added a pull request for the recursive definitions. As you may have seen.
Since so many messages include the same segments, I'm looking for including definitions in different files. I know the JSON schema definition supports that.

I'm working on a definition of a very old, very specific 'dialect', which is probably not useful outside the Netherlands, but whenever I need 'standard' EDIFACT segments, I'll include them. Especially once I get the above suggestion working.

Just curious: have you seen [https://github.com/tdecaluwe/node-edifact](https://github.com/tdecaluwe/node-edifact) ?

**Comment by jono-tt** [comment]
Sorry for the delay, I’m travelling at the moment so haven’t been able to get back to you. Will have a look and let you know if i have any questions. Thanks and appreciate your involvement.

On 25 Jan 2017, at 08:18, hepabolu <notifications@github.com<mailto:notifications@github.com>> wrote:


I've already added a pull request for the recursive definitions. As you may have seen.
Since so many messages include the same segments, I'm looking for including definitions in different files. I know the JSON schema definition supports that.

I'm working on a definition of a very old, very specific 'dialect', which is probably not useful outside the Netherlands, but whenever I need 'standard' EDIFACT segments, I'll include them. Especially once I get the above suggestion working.

Just curious: have you seen https://github.com/tdecaluwe/node-edifact ?
?

**Comment by jono-tt** [comment]
Yes I have seen that one before, the issue is that you need to know edi and it's quirkiness whereas I was trying to abstract that away from the developer so that they just deal with models that are defined by standard schemas.

Out and about

On 25 Jan 2017, at 08:18, hepabolu <notifications@github.com<mailto:notifications@github.com>> wrote:


I've already added a pull request for the recursive definitions. As you may have seen.
Since so many messages include the same segments, I'm looking for including definitions in different files. I know the JSON schema definition supports that.

I'm working on a definition of a very old, very specific 'dialect', which is probably not useful outside the Netherlands, but whenever I need 'standard' EDIFACT segments, I'll include them. Especially once I get the above suggestion working.

Just curious: have you seen https://github.com/tdecaluwe/node-edifact ?
?

-
